### PR TITLE
Clarify what team/* labels are for in PR template

### PR DESCRIPTION
**Pull request by djmitche**
..and move the two QA-related bits of the review checklist together

### CODE REVIEW

**Comment by djmitche** [comment]
(I left this without a `team/*` label to verify my assertion about that label being required even if `qa/skip-qa` is given)

**Review by NouemanKHAL — APPROVED** [review]

**Comment by djmitche** [comment]
image

confirmed  -- we could also think about changing that

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -55,11 +55,11 @@ Note: Adding GitHub labels is only possible for contributors with write access.
 -->
 
 - [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
-- [ ] The appropriate `team/..` label has been applied, if known.
 - [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
 - [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
 - [ ] Changed code has automated tests for its functionality.
 - [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
+- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.  Note that this is required even if `qa/skip-qa` is applied.
```
[the remark this answers is not in the document]
Comment (djmitche) [reply .github/PULL_REQUEST_TEMPLATE.md:62]: > we could also think about changing that

We could change it in another PR..

**Review by mx-psi — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply .github/PULL_REQUEST_TEMPLATE.md:62]: ..ok, that was easier than I thought, so I did it here.

Comment (mx-psi) [inline .github/PULL_REQUEST_TEMPLATE.md:62]:
```suggestion
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
```

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
Removed the label so that we can verify it works

**Pull request merged**